src/metabase/api/search.clj

```diff
@@ -412,28 +383,28 @@
 (mu/defn query-model-set
   "Queries all models with respect to query for one result to see if we get a result or not"
   [search-ctx :- SearchContext]
-  (map #(get (first %) :model)
-       (filter not-empty
-               (for [model search-config/all-models]
-                 (let [search-query     (search-query-for-model model search-ctx)
-                       query-with-limit (sql.helpers/limit search-query 1)]
-                   (mdb.query/query query-with-limit))))))
+  (into #{}
+        (map #(get (first %) :model)
+             (filter not-empty
+                     (for [model search-config/all-models]
+                       (let [search-query     (search-query-for-model model search-ctx)
+                             query-with-limit (sql.helpers/limit search-query 1)]
+                         (mdb.query/query query-with-limit)))))))
```

Comment:
Since this isn't using transducers, you can use `set`. And if you're keen to refactor, this looks too good not to parallelize.
```clojure
(->> search-config/all-models
     (pmap (fn [model]
             (let [search-query     (search-query-for-model model search-ctx)
                   query-with-limit (sql.helpers/limit search-query 1)]
               (mdb.query/query query-with-limit))))
     (filter not-empty)
     (map #(get (first %) :model))
     set)
```

Reply:
I decided to make it even more efficient my using union-all query https://github.com/metabase/metabase/pull/32469/commits/1cdb2db7f22b68e6178cd5d1390bc9e22251a68b